src/js/common/brush.js

```diff
@@ -0,0 +1,95 @@
+function mg_create_brushing_pattern(args, range) {
+  var extentRect;
+  if (d3.select(args.target).select('.mg-extent').size()) {
+    extentRect = d3.select(args.target).select('.mg-extent');
+  } else {
+    extentRect = d3.select(args.target)
+      .select('.mg-rollover-rect, .mg-voronoi')
+      .insert('g', '*')
+      .classed('mg-brush', true)
+      .append('rect')
+      .classed('mg-extent', true);
+  }
+  var x = ('x' in range) ? range.x[0] : args.left;
+  var width = ('x' in range) ? range.x[1] - range.x[0] : args.width - args.left - args.right;
+  var y = ('y' in range) ? range.y[0] : args.top;
+  var height = ('y' in range) ? range.y[1] - range.y[0] : args.height - args.bottom - args.top - args.buffer;
+  extentRect
+    .attr('x', x)
+    .attr('width', width)
+    .attr('y', y)
+    .attr('height', height)
+    .attr('opacity', 1);
+}
+
+function mg_add_event_handler_for_brush(args, target, axis) {
+  var svg = d3.select(args.target).select('svg');
+  var rollover = svg.select('.mg-rollover-rect, .mg-voronoi');
+  var isDragging = false;
+  var mouseDown = false;
+  var origin = [];
+  rollover.classed('mg-brush-container', true);
+  rollover.on('mousedown', function() {
```

Comment:
New functions should be defined using the arrow syntax:

http://airbnb.io/javascript/#arrows--use-them